api/locations/location.mock.ts

```diff
@@ -0,0 +1,42 @@
+/* tslint:disable:max-line-length */
+import MockAdapter from 'axios-mock-adapter';
+import {HttpMockAdapter} from '../common/httpMockAdapter';
+import {ALL_COURT_LOCATIONS} from './data/location.mock.data';
+import {toEpimsLocation} from "./models/location.model";
+
+export const init = () => {
+  const mock: MockAdapter = HttpMockAdapter.getInstance();
+  const getHearingsUrl = /http:\/\/rd-professional-api-aat.service.core-compute-aat.internal\/refdata\/location\/court-venues\/venue-search\?search-string=[\w]*/;
+
+  mock.onGet(getHearingsUrl).reply(config => {
+    const requestURL = config.url;
+    const searchTermStartIdx = config.url.indexOf('=');
+    const searchTermEndIdx = config.url.indexOf('&');
+    let searchTerm;
+
+    if (searchTermEndIdx) {
+      searchTerm = requestURL.substring(searchTermStartIdx + 1, searchTermEndIdx);
+    } else {
+      searchTerm = requestURL.substring(searchTermStartIdx + 1);
+    }
+    const courTypeIdStartIdx = config.url.lastIndexOf('=');
+    const courTypeIdsStr = requestURL.substring(courTypeIdStartIdx + 1);
+    const searchResult = ALL_COURT_LOCATIONS
+      .filter(locationWithCourtType => courTypeIdsStr.indexOf(locationWithCourtType.court_type_id) !== -1)
+      .filter(location =>
+        (location.court_name ? location.court_name.toLowerCase().includes(searchTerm.toLowerCase()) : false)
```

Comment:
I know it's only a mock, but this is a huge OR clause. Could you not have just put these values into a map/array beforehand and then check to see if searchTerm is in that?

Reply:
Merged from https://github.com/hmcts/rpx-xui-webapp/tree/feature/EUI-4690-location-api-to-master - not part of the changes in this PR.